### PR TITLE
First version of the candidatebased GhostTrack tagger

### PR DESCRIPTION
**Pull request by pvmulder**
Similar performance for legacy trackbased GhostTrack tagger (black) and candidatebased taggers on AOD (blue) and miniAOD (red)
roc_gt_765

Agreement between AOD and rerunning from miniAOD reasonable:
gt
Differences to be further investigated.

### CODE REVIEW

File: RecoBTag/SecondaryVertex/src/GhostTrackComputer.cc [hunk 1]
```diff
@@ -316,3 +353,220 @@ GhostTrackComputer::operator () (const TrackIPTagInfo &ipInfo,
...
+		addMeas(*(isTrackVertex ? &tracksDist2D : &vertexDist2D),
+					svInfo.flightDistance(i, true));
+		addMeas(*(isTrackVertex ? &tracksDist3D : &vertexDist3D),
+					svInfo.flightDistance(i, false));
+
+		TrackKinematics &kin = isTrackVertex ? trackKinematics : vertexKinematics;
+
+		for(std::vector<CandidatePtr>::const_iterator track = tracks.begin(); track != tracks.end(); ++track) {
+			kin.add(*(*track)->bestTrack(), 1.0);
+//		for(TrackRefVector::const_iterator track = tracks.begin(); track != tracks.end(); track++) {
```
Comment (slava77) [inline RecoBTag/SecondaryVertex/src/GhostTrackComputer.cc]: here and in other files, please remove commented out code, especially if it's a remainder of now unnecessary debugging step.
If it is needed for some reason, add some comments why, or even compile it and enable with relevant configuration or even compilation switches (ifdef)

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by pvmulder** [comment]
@slava77 I will remove the commented lines

@slava77 @ferencek is runtestPhysicsToolsPatAlgos failing because I should do this PR in 2 stages? First adding it to RECO, only afterwards to PAT? In case so, is it sufficient to comment 'pfGhostTrackBJetTags' in PhysicsTools/PatAlgos/python/producersLayer1/jetProducer_cfi.py? Or also at other locations? 
Also, I have the impression that the AddOn test fails for the same reason as the runtestPhysicsToolsPatAlgos test. Is this correct?

**Comment by ferencek** [comment]
@pvmulder, yes, if the goal is to have the output of this tagger in MiniAOD, then the update needs to be done in 2 stages. For now it should be sufficient to comment `pfGhostTrackBJetTags` in `PhysicsTools/PatAlgos/python/producersLayer1/jetProducer_cfi.py`.

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/14390	515a25d
- code change is in line with the description
- jenkins tests pass and comparisons show no differences (monitored variables are not affected)
- local tests with ttbar PU35 wflow 25202 show that ``` pfGhostTrackVertexTagInfos ``` and ``` pfGhostTrackBJetTags ```  new modules are called run time
    - the CPU time is ~10 ms/evt and ~3 ms/evt, respectively
    - the pfGhostTrackBJetTags and pfGhostTrackVertexTagInfos products show up in RECO output, only the former is in AOD (and it takes ~0.3 kB/evt on this short run; will be less with more events and compression)
    - there appear to be no changes in the miniAOD products; to be applied after relvals are available (held up due to coupling with PAT unit tests, which need available relVal inputs)

**Comment by davidlange6** [comment]
@pvmulder - this PR does not merge now - could you rebase it. Thanks!

**Comment by pvmulder** [comment]
@davidlange6  rebase done.

**Comment by davidlange6** [comment]
please test

**Comment by slava77** [comment]
rebasing to the head of 81X turns out to be a bad idea lately

**Comment by slava77** [comment]
@cmsbuild please test

CMSSW_8_1_X_2016-05-19-1100 is out

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/14390	979991b
- since the last signoff for https://github.com/cms-sw/cmssw/pull/14390	515a25d the code was rebased due to a difference in the context lines
    - as of the previous signoff: https://github.com/cms-sw/cmssw/compare/40be73e3b1ea6bd9d5d0c4712de03aae885da9f5...515a25dd9cc784a3ed4d8a93ecd7767decb05317
    - this version https://github.com/cms-sw/cmssw/compare/01e285615caafc0c4aa36513ef4aabef3ce4656b...979991b26edccf586e63e863affb49bc6518fa71
- jenkins tests pass and comparisons with baseline show no differences (new products are not monitored)

**Pull request merged**